### PR TITLE
feat: new project  landing page

### PR DESCRIPTION
**Pull request by Noggling**
# Description

see cahnge sett :)

- [x] PR title and description are to the point and understandable
- [x] I have performed a self-review of my own code'

Please select version type the purposed change:
- [x] major
- [ ] minor
- [ ] patch
- [ ] none  

External Relations
- [ ] database migration

## Changeset

Project Landing Page Update: Design and User Experience Enhancements
- Enhanced the project landing page by integrating a full-width header containing essential project information, aligning it with the Fusion landing page.
- Introduced a new Overview tab with a new layout.
- Users can now view their project allocation, with a direct link to the project organization application in Fusion.
- Displayed the project director prominently on the project landing page.
- Implemented a project phase indicator on the Overview tab, showcasing DG dates and the current DG phase of the project.
- Aligned the design of pinned apps with Fusion while maintaining content stored in local storage.
- Implemented cleanup functionality for removing deleted apps within pinned apps.
- Redesigned the contract list to feature cards instead of a table, categorizing contracts into active and closed groups based on closing date.
- Introduced a Project Portal Info section with quick facts.
- Implemented a Construction and Commissioning tab featuring milestones and CC-Application KPIs, accessible behind a feature flag.
- Developed a new menu design, also accessible behind a feature flag, in alignment with the all-apps list.
- Added functionality for feature flagging with local storage implementation utilizing the new feature flag module in Fusion.
- Introduced a "My Features" tab under the user's account to enable feature flags.
- Global app search and navigation behind feature flag.

### CODE REVIEW

File: client/packages/components/src/components/app-card/components/ListCard.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		flex: 1;
+	`,
+};
+
+type ListCardProps = {
+	app: Partial<FusionAppManifest>;
+	onClick?: (app: Partial<FusionAppManifest>) => void;
+	onFavorite?: (key: Partial<FusionAppManifest>) => void;
+	loading?: boolean;
+	dark?: boolean;
```
Comment (espenkalle) [inline client/packages/components/src/components/app-card/components/ListCard.tsx:57]: isDark*?

File: client/packages/components/src/components/app-card/components/ListCard.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	pinButton?: boolean;
```
Comment (espenkalle) [inline client/packages/components/src/components/app-card/components/ListCard.tsx:58]: isPinned*

File: client/packages/components/src/components/task/work-assigned/WorkAssigned.tsx [hunk 3]
```diff
@@ -1,43 +1,29 @@
...
-					onClick={() => {
-						actions.setActiveActionById('task');
-					}}
-				>
-					View all
-					<Icon name="chevron_right" />
-				</Button>
-			</Card.Actions>
+			<Styles.Content>
+				<Tasks height={divHeight - 500} />
```
Comment (espenkalle) [inline client/packages/components/src/components/task/work-assigned/WorkAssigned.tsx:25]: Move constant into the hook?

File: client/packages/portal-client/src/components/portal-menu/PortalMenu.tsx [hunk 4]
```diff
@@ -112,6 +127,15 @@ export function MenuGroups() {
 									/>
 								))}
 							</div>
+							{/* Todo: remove when decided to use new menu */}
```
Comment (espenkalle) [inline client/packages/portal-client/src/components/portal-menu/PortalMenu.tsx:130]: Forgot to remove this?

File: client/packages/portal-core/src/context-relations/use-context-relations.ts [hunk 5]
```diff
@@ -13,20 +13,28 @@ export async function getContextRelations(
 	if (!res.ok) throw res;
 	return (await res.json()) as Relations[];
 }
+export async function getContextTstRelations(
+	client: Promise<IHttpClient>,
+	signal?: AbortSignal
+): Promise<Relations[] | undefined> {
+	const res = await (await client).fetch(`/contexts/71db33bb-cb1b-42cf-b5bf-969c77e40931/relations`, { signal });
```
Comment (espenkalle) [inline client/packages/portal-core/src/context-relations/use-context-relations.ts:20]: Hardcoded context?

File: client/packages/portal-pages/src/pages/project-page/Overvirew.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	const { feature: ccTabFeature } = useFrameworkFeature('cc-tab');
+	return (
+		<Styles.Row>
+			<Styles.Col>
+				<WorkAssigned />
+			</Styles.Col>
+			<Styles.Col>
+				<Phases />
+				<Favorites />
+				{/* Todo remove when cc tab is not in feature flag mode */}
```
Comment (espenkalle) [inline client/packages/portal-pages/src/pages/project-page/Overvirew.tsx:23]: remove?

File: client/packages/portal-pages/src/pages/project-page/ProjectPage.Styles.ts [hunk 7]
```diff
@@ -25,10 +25,10 @@ export const StyledContextPageGrid = styled.div`
 	padding: 2rem;
 
 	@media only screen and (min-width: 45rem) and (max-width: 80rem) {
-		grid-template-columns: repeat(6, 1fr);
+		grid-template-columns: repeat(8, 1fr);
 	}
 	@media only screen and (min-width: 80rem) {
-		grid-template-columns: repeat(9, 1fr);
+		grid-template-columns: repeat(8, 1fr);
```
Comment (espenkalle) [inline client/packages/portal-pages/src/pages/project-page/ProjectPage.Styles.ts:31]: Remove condition

File: client/packages/portal-pages/src/pages/project-page/components/app-search/AppSearchBar.tsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+	const matchA = titleNameA.toLowerCase().includes(query.toLowerCase());
+	const matchB = titleNameB.toLowerCase().includes(query.toLowerCase());
+
+	if (matchA && !matchB) return -1;
+	if (!matchA && matchB) return 1;
+	return 0;
+};
+
+export const AppSearchBar = (): JSX.Element => {
```
Comment (espenkalle) [inline client/packages/portal-pages/src/pages/project-page/components/app-search/AppSearchBar.tsx:75]: Duplicated? Is it necessary?

File: client/packages/components/src/components/app-search/AppSearchBar.tsx [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+	const matchA = titleNameA.toLowerCase().includes(query.toLowerCase());
+	const matchB = titleNameB.toLowerCase().includes(query.toLowerCase());
+
+	if (matchA && !matchB) return -1;
+	if (!matchA && matchB) return 1;
+	return 0;
+};
+
+export const AppSearchBar = (): JSX.Element => {
```
Comment (espenkalle) [inline client/packages/components/src/components/app-search/AppSearchBar.tsx:80]: Duplicated? Is it necessary?

File: client/packages/portal-ui/src/portal-menu/group/hooks/useFavorite.ts
Comment (espenkalle) [inline client/packages/portal-ui/src/portal-menu/group/hooks/useFavorite.ts:1]: Empty file.

**Review by espenkalle — APPROVED** [review]

File: client/packages/portal-pages/src/pages/project-page/Overvirew.tsx (hunk 6 above)
Comment (Noggling) [reply client/packages/portal-pages/src/pages/project-page/Overvirew.tsx:23]: nope skal lage issue

File: client/packages/portal-core/src/context-relations/use-context-relations.ts (hunk 5 above)
Comment (Noggling) [reply client/packages/portal-core/src/context-relations/use-context-relations.ts:20]: ahh det ver en test så den skal slettes :)

File: client/packages/portal-client/src/components/portal-menu/PortalMenu.tsx (hunk 4 above)
Comment (Noggling) [reply client/packages/portal-client/src/components/portal-menu/PortalMenu.tsx:130]: nope denne skal inn

Comment (Noggling) [reply client/packages/portal-client/src/components/portal-menu/PortalMenu.tsx:130]: men med issue

File: client/packages/portal-pages/src/pages/project-page/components/app-search/AppSearchBar.tsx (hunk 8 above)
Comment (Noggling) [reply client/packages/portal-pages/src/pages/project-page/components/app-search/AppSearchBar.tsx:75]: yes it is

**Pull request merged**